components/crud-web-apps/jupyter/frontend/src/app/pages/index/index-default/config.ts

```diff
@@ -59,6 +59,11 @@ export const defaultConfig = {
         tooltipField: 'name',
       }),
     },
+    {
+      matHeaderCellDef: 'Type',
+      matColumnDef: 'type',
+      value: new PropertyValue({ field: 'serverType'}),
```

Comment:
Instead of a `PropertyValue` object you can use a [ComponentValue](https://github.com/kubeflow/kubeflow/blob/master/components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/resource-table/types/component-value.ts) instead. In this case you can provide your own Angular component for the specific column.

That component will need to implement the [TableColumn](https://github.com/kubeflow/kubeflow/blob/master/components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/resource-table/component-value/component-value.component.ts#L5-L7) class, which only means that it should have a property `element` in which the table will be injecting the element value of the row. Inside the component you can then do anything you want and show any html [ the svg in this case ] fits your needs.

Here's an example of this 
https://github.com/kubeflow/katib/blob/master/pkg/new-ui/v1beta1/frontend/src/app/pages/experiments/config.ts#L76
https://github.com/kubeflow/katib/blob/master/pkg/new-ui/v1beta1/frontend/src/app/pages/experiments/optimal-trial/experiment-optimal-trial.component.ts

Reply:
Thanks for the instructions. At first I was looking for an example to follow in the `kubeflow/kubeflow` repo but couldn't find any. I had already thought `ComponentValue` would need to be used, but couldn't figure out how to implement this. I'll look into adding this tomorrow. I'll use the same method to dump the SVGs for VSCode and RStudio from the yaml as I am already using so administrators can customize them easily if needed (the VSCode option will likely work for most web-applications).

Reply:
Lets not import svgs into the YAML, I explained the reasoning in the comment below https://github.com/kubeflow/kubeflow/pull/5646#pullrequestreview-614626085.

But lets include the extra links into the `environment.*.ts` files https://github.com/kubeflow/kubeflow/pull/5646#discussion_r596260211

Reply:
I think this is the only thing left to try and implement (the I've already added the SVGs for this as well).